From 911b916fdc53acfc7229302b029fd615f1e670cf Mon Sep 17 00:00:00 2001
From: Babis Chalios <bchalios@amazon.es>
Date: Fri, 23 Aug 2024 12:54:26 +0200
Subject: [PATCH 2/2] ci: adjust 6.1 guest kernel configs for our CI

Set CONFIG_MSDOS_PARTITION=y for our PARTUUID tests,
CONFIG_DEVMEM=y on aarch64 for various integration tests that rely on
/dev/mem being there and CONFIG_IKCONFIG=y, CONFIG_IKCONFIG_PROC=y for
the spectre-meltdown-checker.sh tests to work on AMD.

Signed-off-by: Babis Chalios <bchalios@amazon.es>
---
 .../guest_configs/microvm-kernel-ci-aarch64-6.1.config     | 7 ++++---
 .../guest_configs/microvm-kernel-ci-x86_64-6.1.config      | 7 ++++---
 2 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/resources/guest_configs/microvm-kernel-ci-aarch64-6.1.config b/resources/guest_configs/microvm-kernel-ci-aarch64-6.1.config
index 26b87a65..5c0334f4 100644
--- a/resources/guest_configs/microvm-kernel-ci-aarch64-6.1.config
+++ b/resources/guest_configs/microvm-kernel-ci-aarch64-6.1.config
@@ -140,7 +140,8 @@ CONFIG_RCU_STALL_COMMON=y
 CONFIG_RCU_NEED_SEGCBLIST=y
 # end of RCU Subsystem
 
-# CONFIG_IKCONFIG is not set
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
 # CONFIG_IKHEADERS is not set
 CONFIG_LOG_BUF_SHIFT=17
 CONFIG_LOG_CPU_MAX_BUF_SHIFT=12
@@ -753,7 +754,7 @@ CONFIG_PARTITION_ADVANCED=y
 # CONFIG_AMIGA_PARTITION is not set
 # CONFIG_ATARI_PARTITION is not set
 # CONFIG_MAC_PARTITION is not set
-# CONFIG_MSDOS_PARTITION is not set
+CONFIG_MSDOS_PARTITION=y
 # CONFIG_LDM_PARTITION is not set
 # CONFIG_SGI_PARTITION is not set
 # CONFIG_ULTRIX_PARTITION is not set
@@ -1823,7 +1824,7 @@ CONFIG_HW_RANDOM_VIRTIO=y
 # CONFIG_HW_RANDOM_XIPHERA is not set
 CONFIG_HW_RANDOM_ARM_SMCCC_TRNG=y
 # CONFIG_HW_RANDOM_GRAVITON is not set
-# CONFIG_DEVMEM is not set
+CONFIG_DEVMEM=y
 # CONFIG_TCG_TPM is not set
 # CONFIG_XILLYBUS is not set
 CONFIG_RANDOM_TRUST_CPU=y
diff --git a/resources/guest_configs/microvm-kernel-ci-x86_64-6.1.config b/resources/guest_configs/microvm-kernel-ci-x86_64-6.1.config
index 967e3203..8a2769c0 100644
--- a/resources/guest_configs/microvm-kernel-ci-x86_64-6.1.config
+++ b/resources/guest_configs/microvm-kernel-ci-x86_64-6.1.config
@@ -162,8 +162,9 @@ CONFIG_RCU_STALL_COMMON=y
 CONFIG_RCU_NEED_SEGCBLIST=y
 # end of RCU Subsystem
 
-# CONFIG_IKCONFIG is not set
-# CONFIG_IKHEADERS is not set
+CONFIG_IKCONFIG=y
+CONFIG_IKCONFIG_PROC=y
+# CONFIG_IKHEADERS=y
 CONFIG_LOG_BUF_SHIFT=17
 CONFIG_LOG_CPU_MAX_BUF_SHIFT=12
 CONFIG_PRINTK_SAFE_LOG_BUF_SHIFT=13
@@ -769,7 +770,7 @@ CONFIG_PARTITION_ADVANCED=y
 # CONFIG_AMIGA_PARTITION is not set
 # CONFIG_ATARI_PARTITION is not set
 # CONFIG_MAC_PARTITION is not set
-# CONFIG_MSDOS_PARTITION is not set
+CONFIG_MSDOS_PARTITION=y
 # CONFIG_LDM_PARTITION is not set
 # CONFIG_SGI_PARTITION is not set
 # CONFIG_ULTRIX_PARTITION is not set
-- 
2.34.1

